manim/utils/iterables.py

```diff
@@ -19,12 +19,31 @@
 ]
 
 import itertools as it
-from typing import Any, Callable, Collection, Generator, Iterable, Reversible, Sequence
+from typing import (
+    TYPE_CHECKING,
+    Callable,
+    Collection,
+    Generator,
+    Hashable,
+    Iterable,
+    Reversible,
+    Sequence,
+    TypeVar,
+    overload,
+)
 
 import numpy as np
 
+if TYPE_CHECKING:
+    import numpy.typing as npt
 
-def adjacent_n_tuples(objects: Sequence, n: int) -> zip:
+    T = TypeVar("T")
+    U = TypeVar("U")
+    F = TypeVar("F", np.float_, np.int_)
+    H = TypeVar("H", bound=Hashable)
+
+
+def adjacent_n_tuples(objects: Sequence[object], n: int) -> zip:
```

Comment:
```suggestion
def adjacent_n_tuples(objects: Sequence[T], n: int) -> zip[T]:
```

Reply:
I don't think you can subscript `zip` as it's not a class or a `Protocol`